test/Data/JSON/QQSpec.hs

```diff
@@ -12,39 +12,32 @@ spec :: Spec
 spec = do
   describe "parsedJson" $ do
     it "parses JSON" $ do
-      let Right value = parsedJson "{foo: 23}"
-      value `shouldBe` JsonObject [(HashStringKey "foo", JsonNumber 23)]
+      parsedJson "{foo: 23}" `shouldBe` Right (JsonObject [(HashStringKey "foo", JsonNumber 23)])
 
     it "parses decimal numbers" $ do
-      let Right value = parsedJson "{foo: 5.97}"
-      value `shouldBe` JsonObject [(HashStringKey "foo", JsonNumber 5.97)]
+      parsedJson "{foo: 5.97}" `shouldBe` Right (JsonObject [(HashStringKey "foo", JsonNumber 5.97)])
 
     context "empty objects" $ do
       it "parses empty objects (regression test)" $ do
-        let Right value = parsedJson "{}"
-        value `shouldBe` JsonObject []
+        parsedJson "{}" `shouldBe` Right (JsonObject [])
 
       it "parses empty objects that include whitespace (regression test)" $ do
-        let Right value = parsedJson "{ }"
-        value `shouldBe` JsonObject []
+        parsedJson "{ }" `shouldBe` Right (JsonObject [])
 
       it "parses empty objects that include newlines (regression test)" $ do
-        let Right value = parsedJson "{\n}"
-        value `shouldBe` JsonObject []
+        parsedJson "{\n}" `shouldBe` Right (JsonObject [])
 
     context "empty arrays" $ do
       it "parses empty arrays" $ do
-        let Right value = parsedJson "[   ]"
-        value `shouldBe` JsonArray []
+        parsedJson "[   ]" `shouldBe` Right (JsonArray [])
 
       it "parses empty arrays that include whitespace (regression test)" $ do
-        let Right value = parsedJson "[   ]"
-        value `shouldBe` JsonArray []
+        parsedJson "[   ]" `shouldBe` Right (JsonArray [])
 
       it "parses empty objects that include newlines (regression test)" $ do
-        let Right value = parsedJson "[\n]"
-        value `shouldBe` JsonArray []
+        parsedJson "[\n]" `shouldBe` Right (JsonArray [])
```

Comment:
`parsec` didn't provide `Eq ParseError` in the past (https://github.com/haskell/parsec/blob/master/ChangeLog.md#319). That's the reason why the old code used pattern matching here.

If you want to change this, then please also add `parsec >= 3.1.9` to the test suite.

Reply:
Good to know, thanks. I added this constraint to the test suite.